### DISCUSSION TITLE
how to responsive  SegmentedControl

### DISCUSSION THREAD

**Discussion by goodlythink** [mantinedev/mantine#6246] [Q&A]
when using SegmentedControl with a large amount of data causing it to overflow to the right, you can make it responsive by implementing line wrapping. This would allow the control to automatically adjust its layout, possibly by breaking segments into multiple lines

**Comment by rtivital** [comment]
This is not supported in SegmentedControl component. You can build a custom component with FloatingIndicator on your side.

**Comment by gary-menzel** [comment]
@goodlythink You could try the following:

```jsx
          styles={{
            root: { flexWrap: 'wrap' },
          }}
```
On the SegmentedControl.  This solved my problem with was to make it wrap once sized down to a mobile screen.  It wraps the flex elements as you would normally expect over multiple lines but retains the visibility of the labels.....

Never say "Never".
